include/marl/blockingcall.h

```diff
@@ -32,10 +32,17 @@ class OnNewThread {
   inline static RETURN_TYPE call(F&& f, Args&&... args) {
     RETURN_TYPE result;
     WaitGroup wg(1);
+    auto scheduler = Scheduler::get();
     auto thread = std::thread(
-        [&](Args&&... args) {
-          defer(wg.done());
+        [&, wg](Args&&... args) {
+          if (scheduler != nullptr) {
+            scheduler->bind();
+          }
           result = f(std::forward<Args>(args)...);
+          wg.done();
```

Comment:
TODO: This should probably happen after the call to `unbind` to ensure that all scheduled tasks are completed before the `blocking_call()` returns.

Reply:
Done, uncovering more issues than I was expecting (now fixed by this PR).